cli/flox-rust-sdk/src/models/pkgdb.rs

```diff
@@ -89,7 +93,32 @@ pub enum CallPkgDbError {
 ///
 /// Error JSON is parsed into a [CallPkgDbError::PkgDbError].
 pub fn call_pkgdb(mut pkgdb_cmd: Command) -> Result<Value, CallPkgDbError> {
+    // Configure pkgdb PATH with exact versions of everything it needs.
+    //
+    // Nix itself isn't pure, which is to say that it isn't built with a
+    // reference to `git` in its closure, so correspondingly depends upon
+    // finding it in $PATH to function. Funnily enough it is also not built
+    // to know where `nix` itself resides, and again relies on $PATH for that.
+    //
+    // This just isn't OK for us as we're looking for flox to operate reliably
+    // in "hostile" environments, which includes situation where a user may
+    // redefine or blat their $PATH variable entirely, so we always invoke
+    // pkgdb with an explicit PATH of our making.
+    let pkgdb_paths = [
+        Path::new(&*NIX_PKG_BIN),
+        Path::new(&*GIT_PKG_BIN),
+        // It really shouldn't be necessary to append $PATH,
+        // ... so we won't.
+    ];
+    let pkgdb_path = env::join_paths(pkgdb_paths.iter());
     let mut proc = pkgdb_cmd
+        .env(
+            "PATH",
+            pkgdb_path
+                .unwrap()
+                .to_str()
+                .expect("couldn't convert path to string"),
+        )
```

Comment:
If we don't want to wrap pkgdb, i guess we can at least refactor it to use `expect`.

```suggestion
    let pkgdb_paths = [
        Path::new(&*NIX_PKG_BIN),
        Path::new(&*GIT_PKG_BIN),
        // It really shouldn't be necessary to append $PATH,
        // ... so we won't.
    ];
    let pkgdb_path = env::join_paths(pkgdb_paths.iter())
    	.expect("nix or git bin dir contain invalid characters");
    let mut proc = pkgdb_cmd
        .env("PATH", pkgdb_path)
```


I dont know whether a non-api like this is warranting an error case to handle, would tend to _no_, but if it does we can add a `JoinPath` variant to the `CallPkgDbError` enum and map the join error to that.

```
#[derive(Debug, Error)]
pub enum CallPkgDbError {
    ...
    #[error("could not create value for PATH")]
    JoinPath(#[source] JoinPathsError),
	/// something else error ...
}
    

    let pkgdb_path = env::join_paths(pkgdb_paths.iter())
    	.map_err(CallPkgDbError::JoinPath)?;
```